src/compiler/checker.ts

```diff
@@ -11256,6 +11259,8 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 : declaredType ? declaredType
                 : getAllSymbolFlags(targetSymbol) & SymbolFlags.Value ? getTypeOfSymbol(targetSymbol)
                 : errorType;
+
+            popTypeResolution();

```

Comment:
I would consider writing this like:

```ts
if (!pushTypeResolution() {
	return errorType;
}

let type = somebigComputation()

if (!popTypeResolution() {
	type = errorType;
}

links.type = type
```

Which would match some of the other places we do this, though we usually emit some sort of error.